File: updateReadme.js
const fs = require("fs");

// Path to the README file
const readmeFile = "README.md";

// Get the current year and calculate experience
const startYear = 2021; // Change if your experience started earlier
const yearsOfExperience = new Date().getFullYear() - startYear;

// Read the README file
fs.readFile(readmeFile, "utf8", (err, content) => {
  if (err) {
    console.error("Error reading README.md:", err);
    return;
  }

  // Replace any number before "Years Experience" dynamically
  const updatedContent = content.replace(/\d+ Years Experience/, `${yearsOfExperience} Years Experience`);

  // Write the updated content back to README
  fs.writeFile(readmeFile, updatedContent, "utf8", (err) => {
    if (err) {
      console.error("Error writing to README.md:", err);
      return;
    }
    console.log("✅ README updated successfully!");
  });
});

File: update_readme.py
import datetime
import re

readme_file = "README.md"

# Calculate years of experience dynamically
years_of_experience = datetime.datetime.now().year - 2021
# Read the README content
with open(readme_file, "r", encoding="utf-8") as file:
    content = file.read()

# Replace any number before "Years Experience" dynamically
updated_content = re.sub(r"\d+ Years Experience", f"{years_of_experience} Years Experience", content)

# Write the updated content back
with open(readme_file, "w", encoding="utf-8") as file:
    file.write(updated_content)
